### PR TITLE
bump rust-postgres, proxy: show error context for execution batch execution failure

### PR DESCRIPTION
**Pull request by skyzh**
## Problem

https://github.com/neondatabase/neon/pull/4654#issuecomment-1673059450

## Summary of changes

Add query id and query SQL to the error context. We can have some structured error in the future (i.e., return the ID in JSON instead of within error message). Also refactored `query_to_json` so that it does not consume ownership of the query object.

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: proxy/src/http/sql_over_http.rs [hunk 1]
```diff
@@ -278,13 +285,20 @@ pub async fn handle(
...
-    data: QueryData,
+    data: &QueryData,
     raw_output: bool,
     array_mode: bool,
 ) -> anyhow::Result<Value> {
-    let query_params = json_to_pg_text(data.params)?;
+    let query_params = json_to_pg_text(&data.params)?;
     let row_stream = client
-        .query_raw_txt::<String, _>(data.query, query_params)
+        // TODO: query_raw_txt should be able to accept &str and Vec<String>
```
Comment (conradludgate) [inline proxy/src/http/sql_over_http.rs:294]: done already in my rust-postgres PR

File: proxy/src/http/sql_over_http.rs [hunk 2]
```diff
@@ -278,13 +285,20 @@ pub async fn handle(
...
+        .query_raw_txt::<&str, _>(
+            &data.query,
+            query_params
+                .iter()
+                .map(|x| x.as_ref().map(|y| y.as_str()))
+                .collect::<Vec<Option<&str>>>(),
```
Comment (conradludgate) [inline proxy/src/http/sql_over_http.rs:300]: I think the function accepts iterators, so no need to collect it
```suggestion
```

File: proxy/src/http/sql_over_http.rs (hunk 1 above)
Comment (skyzh) [reply proxy/src/http/sql_over_http.rs:294]: Is https://github.com/neondatabase/rust-postgres/pull/21 ready for merge? I can get it merged and also bump version in this PR

Comment (conradludgate) [reply proxy/src/http/sql_over_http.rs:294]: I'm happy with it. Go ahead

Comment (skyzh) [reply proxy/src/http/sql_over_http.rs:294]: fixed in https://github.com/neondatabase/neon/pull/4958/commits/776cc063c56ad1f10433805a0aa93ac2117d663f

Comment (conradludgate) [reply proxy/src/http/sql_over_http.rs:294]:
```suggestion
```

**Comment by conradludgate** [comment]
I don't like this being a stringly encoded error. @jawj, you can use custom exception types with proper metadata instead of strings, right? I think we should support that instead and add error metadata in a more enriched form rather than as strings

**Pull request closed without merging**